api/requests/serializers.py

```diff
@@ -105,24 +147,62 @@ def create(self, validated_data):
-        request_type = validated_data.pop('request_type', None)
-
-        if request_type != RequestTypes.ACCESS.value:
+        request_type = validated_data.get('request_type', None)
+        if not request_type:
             raise exceptions.ValidationError('You must specify a valid request_type.')
+        elif request_type == NodeRequestTypes.ACCESS.value:
+            return self.make_node_access_request(node, validated_data)
+        elif request_type == NodeRequestTypes.INSTITUTIONAL_REQUEST.value:
+            return self.make_node_institutional_access_request(node, validated_data)
+        else:
+            raise NotImplementedError(f'{request_type} request_type not implemented')
+
+    def make_node_access_request(self, node, validated_data):
+        return self._create_node_request(node, validated_data)
+
+    def make_node_institutional_access_request(self, node, validated_data):
+        node_request = self._create_node_request(node, validated_data)
+        sender = self.context['request'].user
+        message_recipient = OSFUser.load(validated_data.get('message_recipient'))
+        institution_id = validated_data.get('institution')
+        institution = Institution.objects.get(_id=institution_id)
+        if message_recipient:
+            if not message_recipient.is_affiliated_with_institution(institution):
+                raise exceptions.PermissionDenied(f'User {message_recipient._id} not affiliated with institution.')
+            send_mail(
+                to_addr=message_recipient.username,
+                mail=NODE_REQUEST_INSTITUTIONAL_ACCESS_REQUEST,
+                user=message_recipient,
+                **{
+                    'sender': sender,
+                    'recipient': message_recipient,
+                    'comment': validated_data['comment'],
+                    'institution': institution,
+                    'osf_url': settings.DOMAIN,
+                    'node': node_request.target,
+                },
+            )
+        return node_request
 
+    def _create_node_request(self, node, validated_data):
+        creator = self.context['request'].user
+        request_type = validated_data['request_type']
         try:
             node_request = NodeRequest.objects.create(
                 target=node,
-                creator=auth.user,
-                comment=comment,
+                creator=creator,
+                comment=validated_data['comment'],
                 machine_state=DefaultStates.INITIAL.value,
                 request_type=request_type,
+                requested_permissions=validated_data.get('requested_permissions'),
+                message_recipient=OSFUser.load(validated_data.get('message_recipient')),
             )
             node_request.save()
         except IntegrityError:
             raise Conflict(f'Users may not have more than one {request_type} request per node.')
-        node_request.run_submit(auth.user)
+
+        node_request.run_submit(creator)
```

Comment:
Consider that this already sends an email. This PR doesn't cover that that problem, but it will be changed later. This follow will use the state machine to add a curator bool the Contributor class. The follow-up will address "double emailing" for messages and deal with giving the curator role if the request is accepted.